File: Questão-3/ContaBancaria.go
package main

import(
  "fmt"
)

type ContaBancaria struct{
  Titular string
  saldo float64
}

func (c *ContaBancaria)Depositar(valor float64){
  c.saldo += valor
}

func (c *ContaBancaria)Sacar(valor float64){
  if c.saldo < valor{
    fmt.Println("Saldo insuficiente para sacar\n")
  } else{
    c.saldo -= valor
  }
}

func (c ContaBancaria)MostrarSaldo() float64{
  return c.saldo  
}




func main() {
  conta := ContaBancaria{Titular: "João", saldo: 5000}
  conta.Depositar(1000)
  conta.Sacar(7000)
  conta.Sacar(2000)
  fmt.Printf("Saldo da Conta de %s é de %.2f", conta.Titular, conta.MostrarSaldo())
  
  
}

File: Questão-15/Java/contabancaria/ContaBancaria.java
package contabancaria;

public class ContaBancaria {
    private double saldo;
    public String titular;

    //Métodos
    public void depositar(double valor) {
        this.saldo += valor;
    }
    public void sacar(double valor) throws SaldoInsuficiente {
        if (this.saldo < valor) {
            throw new SaldoInsuficiente();
        }
        else{
            this.saldo -= valor;
        }
    }
    //Construtor
    public ContaBancaria(double saldo, String titular) {
        this.saldo = saldo;
        this.titular = titular;
    }

    //Get e sets
    public double getSaldo() {
        return saldo;
    }
    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

    public String getTitular() {
        return titular;
    }

    public void setTitular(String titular) {
        this.titular = titular;
    }
}
